### PR TITLE
[KOGITO-7276] - [KSW-Guides] Building Workflow Images locally with Qu…

### PR DESCRIPTION
**Pull request by spolti**
…arkus CLI

Signed-off-by: spolti <fspolti@redhat.com>

### CODE REVIEW

**Comment by ricardozanini** [comment]
@spolti this is a WIP, right?

**Comment by ricardozanini** [comment]
@spolti can you please rebase?

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 1]
```diff
@@ -1 +1,173 @@
-= Building Workflow Images locally with Quarkus CLI
\ No newline at end of file
+Building Workflow Images locally with Quarkus CLI
+=================================================
+Author Name <fspolti@redhat.com>
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:3]: I think we can remove this. See: https://github.com/kiegroup/droolsjbpm-build-bootstrap#faq

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 2]
```diff
@@ -1 +1,173 @@
...
+v1.0, 06-03-2022
+:compat-mode!:
+// Metadata:
+:description: Build Serverless Application with Quarkus CLI
+:keywords: kogito, workflow, quarkus, serverless, quarkus-cli
+// links
+:quarkus-kogito-examples: https://github.com/kiegroup/kogito-examples/tree/stable/kogito-quarkus-examples
+:example-project: https://github.com/kiegroup/kogito-examples/tree/stable/kogito-quarkus-examples/serverless-workflow-greeting-quarkus
+:quarkus-cli: https://quarkus.io/guides/cli-tooling
+:quarkus-container-images: https://quarkus.io/guides/container-image
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:13]: Pretty cool! cc @akumar074 @amadhusu I think we can add this to the "how to write".

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 3]
```diff
@@ -1 +1,173 @@
...
+
+
+On this guild you will learn how to build a Serverless Application using *Quarkus CLI*.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:16]:
```suggestion
This guide will teach you how to build a Serverless Application image using the *Quarkus CLI*.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 4]
```diff
@@ -1 +1,173 @@
...
+
+== Pre Requisites
+
+Before starting make sure you have the needed dependencies installed.
+For this tutorial we assume that you already have your Serverless Workflow ready to be built inside
+a container image. If not, you can use some of the {quarkus-kogito-examples}[examples], for this tutorial we will be using the {example-project}[serverless-workflow-greeting-quarkus].
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:22]: Wdyt? 

```suggestion
Before starting, make sure you have the needed dependencies installed.
We assume that you already have your Serverless Workflow application ready to be built inside
a container image for this tutorial. If not, you can follow the Getting Started guide, the example project we use in this tutorial.
```

You can then update the links if you agree.

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 5]
```diff
@@ -1 +1,173 @@
...
+
+What will you need:
+
+- Docker - recommended to use always the latest version.
+    * if you don't have it installed or can't install it, don't worry, `JiB` allows you to build Container images wihtouth having the client tooling to build containers, however, it will require the push the built image to a registry, which is described in the following lines.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:27]:
```suggestion
    * It's not a problem if you don't have Docker installed or can't install it. https://github.com/GoogleContainerTools/jib[JiB] allows you to build container images without having the client tooling to build containers. However, Docker is required to push the image to a registry.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 6]
```diff
@@ -1 +1,173 @@
...
+- {quarkus-cli}[Quarkus CLI] {quarkus_version}
+- Java {java_min_version}
+- Maven {maven_min_version}
+- GraalVM {graalvm_min_version}: Note that, GraalVM is option if you already have the expected version installed locally.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:31]: What do you mean here?

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 7]
```diff
@@ -1 +1,173 @@
...
+
+
+== Getting started
+
+First, let's clone the {kogito-examples}[kogito-examples] repository and access the {example-project}[serverless-workflow-greeting-quarkus] example.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:36]:
```suggestion
First, clone the {kogito-examples}[kogito-examples] repository and access the {example-project}[serverless-workflow-greeting-quarkus] example.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 8]
```diff
@@ -1 +1,173 @@
...
+
+.Cloning Kogito Examples
+[source,shell]
+----
+$ git clone --branch stable {kogito-examples} && kogito-examples/kogito-quarkus-examples/serverless-workflow-greeting-quarkus
+----
+
+If you wish to run the example beforehand, follow the steps described on the example.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:44]:
```suggestion
If you wish to run the example beforehand, follow the steps described in the Getting Started guide.
```
If you agree, please add the link to the guide.

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 9]
```diff
@@ -1 +1,173 @@
...
+
+With the example in place, let's visit {quarkus-cli}[Quarkus CLI], follow the installation instructions based on your preferred tool. Make sure you have the following Quarkus version installed:
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:46]: I think this procedure will be covered here: https://github.com/kiegroup/kogito-docs/blob/main/serverlessworkflow/modules/ROOT/pages/getting-started/getting-familiar-with-our-tooling.adoc

So we can skip this explanation now, and add a link to that other guide.

"Assumed that you already have the Quarkus tooling installed, if not please visit: ..."

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 10]
```diff
@@ -1 +1,173 @@
...
+
+=== Checking Quarkus CLI
+
+[source,shell]
+----
+$ quarkus version
+----
+
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:54]: Can we have a section here?

```asciidoc
=== Quarkus Container Extensions
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 11]
```diff
@@ -1 +1,173 @@
...
+.Checkin the JiB extension
+[source,xml,subs="attributes+"]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-container-image-jib</artifactId>
+</dependency>
+----
+
+=== Building the Serverless Application
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:77]:
```suggestion
=== Building the Serverless Workflow Application
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 12]
```diff
@@ -1 +1,173 @@
...
+    -Dquarkus.container-image.tag=1.0 \
+    -Dquarkus.container-image.registry=quay.io
+----
+
+Note that JiB caches the `target/lib`, that, depending on how big your project is, it can speed up the rebuilds.
+If you want to push it to a registry, make sure to set the following properties accordingly your needs:
+
+- quarkus.container-image.registry: registry address, e.g. _quay.io_.
+- quarkus.container-image.group: registry _context/namespace_ or if on k8s or ocp the _namespace/project_.
+- quarkus.container-image.push: needs to be set to `true`, identify whether to push or not the image, defaults to `false`.
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:93]: What about suggesting the user to create a maven profile. See: https://github.com/kiegroup/kogito-examples/blob/main/kogito-quarkus-examples/serverless-workflow-newsletter-subscription/subscription-flow/pom.xml#L215-L227

Can be an additional, optional, more advanced section.

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 13]
```diff
@@ -1 +1,173 @@
...
+    -Dquarkus.native.container-build=true
+----
+
+The command above will result on the following Container image: `quay.io/spolti/serverless-workflow-greeting-quarkus:1.0-native`
+
+If you don't have GraalVM installed, you can specify the _**quarkus.native.container-build=true**_ system properties. This property will create a Linux executable. Notice that, Native builds take longer than usual java builds.
+
+=== Testing the built Container Image
+
+If you do have Docker installed, the image will also be pushed to Docker's local registry:
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:130]:
```suggestion
If you do have Docker installed, Quarkus will push the image to Docker's local registry:
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 14]
```diff
@@ -1 +1,173 @@
...
+----
+$ docker run -it -p 8080:8080 quay.io/spolti/serverless-workflow-greeting-quarkus:1.0
+...
+INFO  [io.quarkus] (main) serverless-workflow-greeting-quarkus 1.22.1.Final on JVM (powered by Quarkus 2.9.2.Final) started in 1.302s
+----
+
+.Native Image
+[source,shell]
+----
+$ docker run -it -p 8080:8080 quay.io/spolti/serverless-workflow-greeting-quarkus:1.0-native
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:143]:
```suggestion
$ docker run -it -p 8080:8080 quay.io/kogito/serverless-workflow-greeting-quarkus:1.0-native
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 15]
```diff
@@ -1 +1,173 @@
...
+...
+INFO  [io.quarkus] (main) serverless-workflow-greeting-quarkus 1.22.1.Final native (powered by Quarkus 2.9.2.Final) started in 0.039s
+----
+
+Notice that the Native image startup is more than 1 second faster than the first one.
+Let's invoke the application:
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:149]:
```suggestion
Let's invoke the application.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 16]
```diff
@@ -1 +1,173 @@
...
+
+With the native image started, execute the following command:
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:151]:
```suggestion
Having the image started, execute the following command:
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 17]
```diff
@@ -1 +1,173 @@
...
+
+[source,shell]
+----
+$ curl -X POST -H 'Content-Type:application/json' -H 'Accept:application/json' -d '{"workflowdata" : {"name": "John", "language": "English"}}' http://localhost:8080/jsongreet
+----
+
+As output, the following response will be returned:
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:158]:
```suggestion
Check to see if the response is similar to the following:
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 18]
```diff
@@ -1 +1,173 @@
...
+{
+    "id": "2acf710d-7e4a-481d-925c-dfd85a369987",
+    "workflowdata": {
+        "name": "John",
+        "language": "English",
+        "greeting": "Hello from JSON Workflow, "
+    }
+}
+----
+
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:170]: Maybe we can add a section "What's Next" and add a few more links. I believe the next step will be deploying to Minikube (and also the k8s one you added).

**Review by ricardozanini — COMMENTED** [review]
Great work, @spolti! I left a few suggestions if you don't mind. Cheers!

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc (hunk 6 above)
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:31]: rephrased.

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc (hunk 9 above)
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:46]: +1

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:90]: that is a good idea.

[the remark this answers is not in the document]
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:90]: Actually, I am not sure about it, I didn't find a way to select the maven profile with quarkus cli.

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc (hunk 18 above)
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:170]: Added.
So far, minikube and kubernetes.

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (ricardozanini) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:93]: Oh yes, the user would have to change the pom manually.

[the remark this answers is not in the document]
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:93]: added.

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 19]
```diff
@@ -1 +1,202 @@
...
+// links
+:quarkus-kogito-examples: https://github.com/kiegroup/kogito-examples/tree/stable/kogito-quarkus-examples
+:example-project: https://github.com/kiegroup/kogito-examples/tree/stable/kogito-quarkus-examples/serverless-workflow-greeting-quarkus
+:quarkus-cli: https://quarkus.io/guides/cli-tooling
+:quarkus-container-images: https://quarkus.io/guides/container-image
+:quarkus-native-builds: https://quarkus.io/guides/building-native-image
+:google-jib: https://github.com/GoogleContainerTools/jib
+
+
+This guide will teach you how to build a Serverless Application Container Image using the *Quarkus CLI*.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:17]: Can we add the link to the Quarkus CLI?

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 20]
```diff
@@ -1 +1,202 @@
...
+
+== Pre Requisites
+
+Before starting, make sure you have the needed dependencies installed.
+We assume that you already have your Serverless Workflow application ready to be built inside
+a container image for this tutorial. If not, you can follow the Getting Started guide, the example project we use in this tutorial.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:23]: Can we have the link for the getting started guide?

**Review by ricardozanini — APPROVED** [review]
LGMT! Can you please incorporate your changes to the getting started adoc as well?

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 21]
```diff
@@ -1 +1,202 @@
...
+
+If you wish to run the example beforehand, follow the steps described in the Getting Started guide.
+
+With the example in place, let's visit {quarkus-cli}[Quarkus CLI], follow the installation instructions based on your preferred tool. Make sure you have the following Quarkus version installed:
+
+=== Quarkus Container Extensions
+
+At this point it is ssumed that you already have the Quarkus tooling installed, if not please visit: xref:getting-started/getting-familiar-with-our-tooling.adoc[Getting familiar with our tooling].
+
+Quarkus provides a few extensions to build container images, like `JiB`, `docker`, `s2i`, and `buildpacks`.
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:54]:
```suggestion
Quarkus provides a few extensions to build container images, like `Jib`, `docker`, `s2i`, and `buildpacks`. +
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 22]
```diff
@@ -1 +1,202 @@
...
+For this guide, we use `JiB`. However, if you already use another extension or want to know how it works, please visit this {quarkus-container-images}[Quarkus' guide].
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:55]: Please replace all `JiB` with `Jib`. See https://github.com/GoogleContainerTools/jib

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 23]
```diff
@@ -1 +1,202 @@
...
+
+.Adding the JiB extension
+[source,shell]
+----
+$ quarkus extension add 'container-image-jib'
+----
+
+If you inspect the pom.xml, you can check the newly added dependency.
+
+.Checkin the JiB extension
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:68]:
```suggestion
.Checking the JiB extension
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 24]
```diff
@@ -1 +1,202 @@
...
+    <quarkus.container-image.group>kogito</quarkus.container-image.group>
+    <quarkus.container-image.name>serverless-workflow-greeting-quarkus</quarkus.container-image.name>
+    <quarkus.container-image.tag>1.0</quarkus.container-image.tag>
+    <quarkus.container-image.registry>quay.io</quarkus.container-image.registry>
+    <quarkus.container-image.push>true</quarkus.container-image.push>
+  </properties>
+</profile>
+----
+
+If using profiles, a new extension can be added to the target profile by setting profile name like the example below:
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:126]:
```suggestion
If you are using profiles, a new extension can be added to the targeted profile by setting the profile name in the build command like the example below:
```

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 25]
```diff
@@ -1 +1,202 @@
...
+    -Dquarkus.container-image.group=kogito \
+    -Dquarkus.container-image.name=serverless-workflow-greeting-quarkus \
+    -Dquarkus.container-image.tag=1.0-native \
+    -Dquarkus.container-image.registry=quay.io \
+    -Dquarkus.native.container-build=true
+----
+
+The command above will result in the following container image: `quay.io/kogito/serverless-workflow-greeting-quarkus:1.0-native`
+
+If you don't have GraalVM installed, you can specify the _**quarkus.native.container-build=true**_ system properties. This property will create a Linux executable. Notice that, Native builds take longer than usual java builds. For more information about native builds please visit this {quarkus-native-builds}[guide].
```
Comment (radtriste) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:152]:
```suggestion
NOTE: If you don't have GraalVM installed, you can specify the _**quarkus.native.container-build=true**_ system properties. This property will create a Linux executable. Notice that, Native builds take longer than usual java builds. For more information about native builds please visit this {quarkus-native-builds}[guide].
```

File: antora-playbook-author.yml [hunk 26]
```diff
@@ -19,4 +19,7 @@ asciidoc:
     linkattrs: ''
     toc: ~
     quarkus_version: '2.9.1.Final'
-    required_maven_version: '3.8.1+'
\ No newline at end of file
+    java_min_version: 11+
+    maven_min_version: 3.8.1+
+    graalvm_min_version: 22.0.0.2
```
[the remark this answers is not in the document]
Comment (spolti) [reply antora-playbook-author.yml:24]: I don't know
@cristianonicolai @ricardozanini do you guys know?

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 27]
```diff
@@ -1 +1,202 @@
...
+.Using Quarkus CLI with Maven profile
+[source,shell]
+----
+$ quarkus build -- -Pbuild-container
+----
+
+
+==== Building the Serverless Workflow Application with Native Image
+
+And what about native builds? When we talk about Serverless, we expect a small startup footprint that can be better when using Native Builds. Quarkus CLI allows you to do it by passing the _--native_ flag:
```
[the remark this answers is not in the document]
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:136]: is the + signal correct?

[the remark this answers is not in the document]
Comment (radtriste) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:136]: https://docs.asciidoctor.org/asciidoc/latest/syntax-quick-reference/#paragraphs see `hard line breaks`

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc (hunk 24 above)
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:125]: This sentence should be rephrased, there is no support for adding extension with cli or quarkus maven plugin.

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 28]
```diff
@@ -1 +1,202 @@
...
+:google-jib: https://github.com/GoogleContainerTools/jib
+
+
+This guide will teach you how to build a Serverless Application Container Image using the {quarkus-cli}[Quarkus CLI].
+
+== Pre Requisites
+
+Before starting, make sure you have the needed dependencies installed.
+We assume that you already have your Serverless Workflow application ready to be built inside
+a container image for this tutorial. If not, you can follow the Getting Started xref:getting-started/create-your-first-workflow-service.adoc[guide], of use the example project we've use in this tutorial.
```
[the remark this answers is not in the document]
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:22]: is this + signal correct?

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 29]
```diff
@@ -1 +1,202 @@
...
+
+
+== Getting started
+
+First, clone the {kogito-examples}[kogito-examples] repository and access the {example-project}[serverless-workflow-greeting-quarkus] example.
+
+.Cloning Kogito Examples
+[source,shell]
+----
+$ git clone --branch stable {kogito-examples} && kogito-examples/kogito-quarkus-examples/serverless-workflow-greeting-quarkus
```
[the remark this answers is not in the document]
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:41]: updated.

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 30]
```diff
@@ -1 +1,202 @@
...
+----
+
+If you wish to run the example beforehand, follow the steps described in the Getting Started guide.
+
+With the example in place, let's visit {quarkus-cli}[Quarkus CLI], follow the installation instructions based on your preferred tool. Make sure you have the following Quarkus version installed:
```
[the remark this answers is not in the document]
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:46]: fixed,

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 31]
```diff
@@ -17,9 +17,9 @@ This guide will teach you how to build a Serverless Application Container Image
 
 == Prerequisites
 
-Before starting, make sure you have the needed dependencies installed.
+Before starting, make sure you have the needed dependencies installed. +
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:20]:
```suggestion
Before starting, make sure you have the needed dependencies installed. +
```

Just curious, why do we need this plus sign? What does it do?

Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:20]: I made the same question to Tristan :)
Yes.

Comment (radtriste) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:20]: this is forcing a line break: https://docs.asciidoctor.org/asciidoc/latest/syntax-quick-reference/#paragraphs

**Comment by ricardozanini** [comment]
@davidesalerno @ibek @radtriste can you please take a look before handling it over to @hmanwani-rh ?

File: antora-playbook-author.yml (hunk 26 above)
[the remark this answers is not in the document]
Comment (ricardozanini) [reply antora-playbook-author.yml:24]: Not sure, I think we will need to update these versions manually. Or we scan Quarkus' pom.xml or ascii doc files.

[the remark this answers is not in the document]
Comment (ricardozanini) [reply antora-playbook-author.yml:24]: @spolti since you changed the variable's names here, can you also please update the naming on the getting started page?

[the remark this answers is not in the document]
Comment (spolti) [reply antora-playbook-author.yml:24]: Sure.

[the remark this answers is not in the document]
Comment (spolti) [reply antora-playbook-author.yml:24]: done

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 32]
```diff
@@ -1 +1,209 @@
...
+[source,shell,subs="attributes+"]
+----
+$ git clone --branch stable {kogito_examples} && cd kogito-examples/kogito-quarkus-examples/serverless-workflow-greeting-quarkus
+----
+
+If you wish to run the example beforehand, follow the steps described in the Getting Started guide.
+
+With the example in place, let's visit {quarkus-cli}[Quarkus CLI], follow the installation instructions based on your preferred tool. Make sure you have the following Quarkus version installed:
+
+.Adding the Jib extension
```
Comment (akumar074) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:48]: This line looks unrelated here.

**Review by akumar074 — CHANGES_REQUESTED** [review]
@spolti Thanks for the PR. Please take a look at the comment.

**Review by ricardozanini — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 33]
```diff
@@ -1 +1,209 @@
...
+:compat-mode!:
+// Metadata:
+:description: Build Serverless Application with Quarkus CLI
+:keywords: kogito, workflow, quarkus, serverless, quarkus-cli
+// links
+:example-project: https://github.com/kiegroup/kogito-examples/tree/stable/kogito-quarkus-examples/serverless-workflow-greeting-quarkus
+
+:quarkus-container-images: https://quarkus.io/guides/container-image
+:quarkus-native-builds: https://quarkus.io/guides/building-native-image
+:google-jib: https://github.com/GoogleContainerTools/jib
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:13]: I think we need to standardize these variables by adding a suffix to them. For example: `google-jib-url`

File: antora-playbook.yml [hunk 34]
```diff
@@ -19,4 +19,8 @@ asciidoc:
     linkattrs: ''
     toc: ~
     quarkus_version: '2.9.1.Final'
-    required_maven_version: '3.8.1+'
\ No newline at end of file
+    quarkus_cli: https://quarkus.io/guides/cli-tooling
+    java_min_version: 11+
+    maven_min_version: 3.8.1+
+    graalvm_min_version: 22.0.0.2
+    kogito_examples: https://github.com/kiegroup/kogito-examples.git
```
Comment (ricardozanini) [inline antora-playbook.yml:26]: Are we using `_` or `-`? Your local variables are using `-`. I'm fine with whatever we decide, but we need to standardize, IMO.

Sorry for these late comments, but now that I'm writing another guide, I realized these small nits.

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc (hunk 33 above)
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:13]: sounds good.
@akumar074 please add this to the contributors' guide.

File: antora-playbook.yml (hunk 34 above)
Comment (spolti) [reply antora-playbook.yml:26]: `_` I suppose, will update it.

**Review by ricardozanini — APPROVED** [review]

**Review by akumar074 — APPROVED** [review]
LGTM Thanks @spolti

**Review by amadhusu — APPROVED** [review]

**Review by krisv — CHANGES_REQUESTED** [review]

**Review by davidesalerno — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 35]
```diff
@@ -1 +1,208 @@
...
+----
+$ quarkus build -Dquarkus.container-image.build=true \
+    -Dquarkus.container-image.group=kogito \
+    -Dquarkus.container-image.name=serverless-workflow-greeting-quarkus \
+    -Dquarkus.container-image.tag=1.0 \
+    -Dquarkus.container-image.registry=quay.io
+----
+
+Note that Jib caches the `target/lib` directory. Depending on how big your project is, it can speed up the rebuilds.
+If you want to push it to a registry, make sure to set the following properties accordingly to your needs:
```
[the remark this answers is not in the document]
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:93]: there will be a another topic about it.

[the remark this answers is not in the document]
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:93]: It is just a example that illustrate how to use Jib to push the built container automatically, the steps required to push to minikube and k8s will be better described on the topics listed on the What's next section.

File: serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc [hunk 36]
```diff
@@ -1 +1,208 @@
...
+    -Dquarkus.container-image.tag=1.0 \
+    -Dquarkus.container-image.registry=quay.io \
+    -Dquarkus.container-image.push=true
+----
+
+The command above will result in the following container image: `quay.io/kogito/serverless-workflow-greeting-quarkus:1.0`.
+
+Alternatively, you can create a maven profile to build the container image that can be triggered just by setting the target profile.
+
+.Maven Profile Example
```
[the remark this answers is not in the document]
Comment (spolti) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:114]: @ricardozanini have asked for it :)

[the remark this answers is not in the document]
Comment (krisv) [reply serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc:116]: I'm fine either way.  I think it's probably something we want to revisit if this could be done easier in the future.

**Review by krisv — APPROVED** [review]

**Review by hmanwani-rh — APPROVED** [review]
@spolti I've made changes to the content of the guide. Please take a look and let me know if the changes look good.

**Review by spolti — COMMENTED** [review]
sounds good to me, thanks @hmanwani-rh

**Comment by hmanwani-rh** [comment]
> sounds good to me, thanks @hmanwani-rh

@spolti Here's the doc preview: https://deploy-preview-8--tangerine-crostata-04a6bf.netlify.app/serverlessworkflow/main/cloud/build-workflow-image-with-quarkus-cli.html
